### PR TITLE
Do not run release_build targets using cocoon scheduler.

### PR DESCRIPTION
**Pull request by godofredoc**
Targets with the release_build property are producing release artifacts and when running from release candidate branches it should run exclusively from dart_internal.

Bug: https://github.com/flutter/flutter/issues/128847

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by godofredoc** [comment]
@XilaiZhang @Jasguerrero this is the PR releasing engine v2 artifacts in dart-internal. Please let me know if you have questions.

File: app_dart/test/model/ci_yaml/ci_yaml_test.dart [hunk 1]
```diff
@@ -176,6 +176,35 @@ void main() {
         );
       });
 
+      test('filter release_build targets from release candidate branches', () {
+        final CiYaml releaseYaml = CiYaml(
+          slug: Config.flutterSlug,
+          branch: 'flutter-2.4-candidate.3',
```
Comment (Jasguerrero) [inline app_dart/test/model/ci_yaml/ci_yaml_test.dart:182]: Can you add a test for a branch that dosen't contain `flutter`, also one for `flutter-3.10-candidate.1` this way once we delete that part of the code we have also a test to change/delete

**Review by Jasguerrero — CHANGES_REQUESTED** [review]

File: app_dart/test/model/ci_yaml/ci_yaml_test.dart (hunk 1 above)
Comment (godofredoc) [reply app_dart/test/model/ci_yaml/ci_yaml_test.dart:182]: Updated!

**Review by XilaiZhang — APPROVED** [review]
LGTM!
images (1)

**Review by Jasguerrero — APPROVED** [review]
LGTM

**Pull request merged**